Eatery/Controllers/ImageParallaxScrollViewController.swift

```diff
@@ -17,107 +18,116 @@ import UIKit
             make.height.equalTo(view).dividedBy(3)
             make.top.leading.trailing.width.equalToSuperview()
         }
+    }
 
-        navigationBar.tintColor = .white
-        navigationBar.setBackgroundImage(UIImage(), for: .default)
-        navigationBar.barTintColor = .blue
-        navigationBar.shadowImage = UIImage()
-        navigationBar.delegate = self
-        scrollView.addSubview(navigationBar)
-        navigationBar.snp.makeConstraints { make in
-            make.top.equalToSuperview()
-            make.leading.trailing.equalToSuperview()
-        }
-
-        let gradientView = GradientView(frame: .zero)
-        gradientView.alpha = 0.5
+    private func setUpGradientView() {
         scrollView.addSubview(gradientView)
         gradientView.snp.makeConstraints { make in
             make.leading.trailing.equalToSuperview()
             make.bottom.equalTo(imageView)
             make.height.equalTo(imageView).multipliedBy(2.0 / 3.0)
         }
+    }
 
-        titleLabel.isOpaque = false
-        titleLabel.font = .boldSystemFont(ofSize: 34)
-        titleLabel.numberOfLines = 2
-        titleLabel.textColor = .white
-        titleLabel.setContentCompressionResistancePriority(UILayoutPriority(rawValue: 100), for: .horizontal)
-        scrollView.addSubview(titleLabel)
-        titleLabel.snp.makeConstraints { make in
-            make.leading.equalToSuperview().inset(16)
-            make.bottom.equalTo(imageView.snp.bottom).inset(16)
-        }
-
-        scrollView.addSubview(accessoryView)
-        accessoryView.snp.makeConstraints { make in
-            make.leading.equalTo(titleLabel.snp.trailing).offset(16)
-            make.width.equalTo(0).priority(.low) // compress the view if its not needed
-            make.trailing.equalToSuperview()
-            make.bottom.equalTo(imageView.snp.bottom).inset(16)
+    private func setUpHeaderView() {
+        scrollView.addSubview(headerView)
+        headerView.snp.makeConstraints { make in
+            make.top.bottom.equalTo(imageView)
+            make.leading.trailing.width.equalToSuperview()
         }
+    }
 
-        contentView.backgroundColor = .green
+    private func setUpContentView() {
         scrollView.addSubview(contentView)
         contentView.snp.makeConstraints { make in
             make.top.equalTo(imageView.snp.bottom)
-            make.leading.trailing.bottom.equalToSuperview()
+            make.leading.trailing.bottom.width.equalToSuperview()
         }
+    }
+
+    private func setUpNavigationBarAndBackground() {
+        navigationBar.tintColor = .white
+        navigationBar.setBackgroundImage(UIImage(), for: .default)
+        navigationBar.shadowImage = UIImage()
+        navigationBar.delegate = self
+
+        view.addSubview(navigationBar)
+        navigationBar.snp.makeConstraints { make in
+            make.top.equalTo(view.snp.topMargin)
+            make.leading.trailing.equalToSuperview()
+        }
```

Comment:
not sure what is more consistent in eatery but I prefer doing:
```
let newView = UIView()
...
view.addSubview(newView)

newView.snp.makeConstraints{}
```

Reply:
Is, this not what's going on?

Reply:
oh i meant as in move line 95 to line 94.